src/test/java/com/cdancy/bitbucket/rest/features/TasksApiLiveTest.java

```diff
@@ -121,15 +129,34 @@ public void testGetTaskOnError() {
         assertThat(instance).isNotNull();
         assertThat(instance.errors().isEmpty()).isFalse();
     }
-    
+
+    @Test
+    public void testTaskStatusUpdate() {
+        final CreateTask createTask = CreateTask.create(commentId, taskComment);
+        final Task instanceAlreadyOpen = api().create(createTask);
+        final TaskAnchor anchor = instanceAlreadyOpen.anchor();
+
+        final UpdateTask updateTaskAlreadyOpen = UpdateTask.update(anchor, instanceAlreadyOpen.id(), TASK_RESOLVED, pullRequestId, repositoryId);
+        final Task instanceNowResolved = api().update(updateTaskAlreadyOpen.id(), updateTaskAlreadyOpen);
+        assertThat(instanceNowResolved).isNotNull();
+        assertThat(instanceNowResolved.errors().isEmpty()).isTrue();
+        assertThat(instanceNowResolved.state()).isEqualTo("RESOLVED");
+
+        final UpdateTask updateTaskBackToOpen = UpdateTask.update(anchor, instanceNowResolved.id(), TASK_OPEN, pullRequestId, repositoryId);
+        final Task instanceReopened = api().update(updateTaskBackToOpen.id(), updateTaskBackToOpen);
```

Comment:
The `.update` method signature is a bit weird with passing `updateTaskBackToOpen.id()` _and_ the object `updateTaskBackToOpen` - normally I'd just pass the object. But this is required because of how `.update` maps to the Bitbucket API (out of our control). Maybe a wrapper method could be made instead but that was not in keeping with how the rest of the API is being done.

Reply:
Yeah it is a bit weird and hate when api's do this type of thing. We _could_ simplify things and have it work as expected just not sure if making this more complicated is a good idea. Let me think about things overnight and get back to you tomorrow.

Reply:
Actually ... what happens if you live either one out? I suspect the pathParam is required but what happens if you leave things out of the object?

Reply:
You mean leave out the object completely? So, for TaskApi, make it 

```
Task update(@PathParam("taskId") int taskId);
```

Reply:
So when I look at the documentation for this task the body of the response can only be:
```
{
    "id": 99,
    "text": "Fix the missing and obsolete imports"
}
```
It looks like you're adding in a lot more properties with the `UpdateTask` class.

Reply:
Good point, I have removed the pull request and repository ID properties.